tests/contrib.print_proxy.mapfish_print/resources/expected_getspec_extract.json

```diff
@@ -521,7 +521,7 @@
       "Theme_Text": "Nutzungsplanung: 08.73B.Ueberlagerung_Flaeche",
       "Theme_Code": "ch.Nutzungsplanung",
       "SubTheme_Code": "ch.Nutzungsplanung",
```

Comment:
I cannot figure out where the SubTheme item inside the  RestrictionOnLandownership comes from https://github.com/openoereb/pyramid_oereb/pull/1468/files/e64746952032242c2df74421a97f5d80143cd24a#diff-4c4e9510d3963c2577b99fc177665ed64d5ae65333e00c770afe101c5dbb58f0R908

In the legacy test fixtures, there is an item SubTheme, but I cannot find any part of code with would create a 'SubTheme' section from the DB. Also on a running oereb server with dev data, I cannot obtain a SubTheme section.

Therefore the concatenation here will have no effect except for test cases: https://github.com/openoereb/pyramid_oereb/blob/1430-fix-mfp-subtheme-splitting/pyramid_oereb/contrib/print_proxy/mapfish_print/mapfish_print.py#L374

Reply:
It would be more accurate to do the title concatenation in the json generation here https://github.com/openoereb/pyramid_oereb/pull/1432/files#diff-f8a6031e0f7b5b34937fcc2383b533fef4e2c2f6fc73f48a0da1f0704768cc7fR472

rather than in the MFP preparation, but there have been long discussions whether to concatenate or not.

Reply:
There may be relics from the old code in here. In the old JSON-extract we had three attributes concerning themes and subthemes:
* `Theme.Code`: the theme's code (e.g. ch.Nutzungsplanung)
* `Theme.Text`: the theme's text (e.g. Kommunale Nutzungsplanung)
* `SubTheme`: the subtheme's text (e.g. Linienbezogene Festlegungen)

![old_json](https://user-images.githubusercontent.com/3659322/149501248-f882842b-c358-403e-a05f-607b9cdb8b96.png)

In the new extract we have:
* `Theme.Code`: the theme's code (e.g. ch.Nutzungsplanung)
* `Theme.SubCode`: the subtheme's code (e.g.ch.Nutzungsplanung.Linie)
* `Theme.Text`: the theme's text (e.g. Kommunale Nutzungsplanung)

There is only one text field so there's no concatenation possible.

Reply:
> It would be more accurate to do the title concatenation in the json generation here https://github.com/openoereb/pyramid_oereb/pull/1432/files#diff-f8a6031e0f7b5b34937fcc2383b533fef4e2c2f6fc73f48a0da1f0704768cc7fR472
> 
> rather than in the MFP preparation, but there have been long discussions whether to concatenate or not.

I agree with you and it would be ok for me to concatenate in the json generator.

Reply:
One could even consider this as a flaw in the spec. There should be four attributes in the json:
* Theme.Code
* Theme.SubCode
* Theme.Text
* Theme.SubText

This way any client could do what he wants (concatenate etc.).

Reply:
thanks a lot for this explanation, this corresponds to my analysis, but I did not have the complete understanding.

I have summed up the issue in the discussion thread here https://github.com/openoereb/pyramid_oereb/issues/1467#issuecomment-1013010449
otherwise there will be too much cross discussion.

I can implement a suggestion for concatenation in the json generation, so everybody can try it out.

Reply:
@vvmruder it looks like some legacy code to me. Especially if it is only in the test cases.
From what I remember the following is specified/to be used:

- ` Theme.Code` Every Theme has this also sub themes
- `Theme.SubCode` Only themes with a sub theme have this to identify the sub.theme
- `Theme.Text` Every theme has this.

So in fact a sub theme is a theme on its own but is added to the theme through the theme.subCode and then can be fetched through that again (I hope I remember that correctly). Which makes this `SubTheme_Code`.
Also the trick was to start of with the legends the "belong" to a theme or sub-them and the sub theme is in the hierarchy below the theme.

Reply:
I think Clemens' comment was on a previous commit.
In the last commits I changed SubTheme_Code into Theme_SubCode.
I will therefore be able to merge the PR.